### DISCUSSION TITLE
How do you batch convert multiple webp into multiple gifs

### DISCUSSION THREAD

**Discussion by 0196071** [ImageMagick/ImageMagick#7151] [Help] [answered]
### ImageMagick version

7.0.1+29

### Operating system, version and so on

Windows 11 22h2

### Description

How do you batch convert multiple webp into multiple gifs. I cant seem to find a way to do it.

**Comment by fmw42** [comment, marked as answer]
Have you tried magick mogrify?

```
Put the webp files that you want to convert into a new folder, i.e. directory
cd to directory with webp files
magick mogrify -format gif *.webp

```

**Reply by 0196071** [reply]
for now its seems to work but is there any ways to do it with convert?

**Reply by fmw42** [reply]
Why do you have to use magick (convert is not recommended in IM 7)?  If you have to use convert or magick, then you would have to write a script FOR loop over each image and transform it one image at a time.

**Reply by 0196071** [reply]
must be because any info i've found about webp to gif dated from at least 3 years ago

**Reply by 0196071** [reply]
there was a bunch of complains about artifacts with mogrify

**Reply by 0196071** [reply]
hmm 4 years ago to be exact https://stackoverflow.com/questions/59977561/how-can-i-avoid-dropouts-when-using-imagemagick-mogrify-to-convert-webp-file

**Comment by fmw42** [comment]
Their version of IM 6 in the link was old and likely buggy.  I know of no current issue with current versions of IM 7 magick mogrify.

**Comment by ramdany7** [comment]
i dont know if you can do it with only "convert.exe" or not, but you definitely can do it using a batch script.

```batch
@Echo off
cd /d %~dp0
set "Convert=D:\C:\Program Files\ImageMagick\convert.exe"
set "Source=.webp"
set "Destination=.gif"

rem Searching *.gif in current folder and subfolders
for /r %%I in (*%Source%) do (
    set "FilePath=%%~dpI"
    set "InputFile=%%~nxI"
    set "OutputFile=%%~nI%Destination%"
    call :Proceed
)
echo.
echo Done.
pause>nul&exit


:Proceed
cd /d "%FilePath%"

rem Displaying file path
if /i not "%FilePath%"=="%CurrentPath%" (
    set "CurrentPath=%FilePath%"
    echo. 
    echo ^[%FilePath%^]
)

rem Displaying file name
echo - %InputFile%

rem Converting the file
"%Convert%" "%InputFile%" "%OutputFile%"

if exist "%OutputFile%" echo - %OutputFile% (Success!)
if not exist "%OutputFile%" echo - %OutputFile% (Fail!)
exit /b
```
- Change this line to your **Convert.exe** path
`set "Convert=D:\C:\Program Files\ImageMagick\convert.exe"`
- Remove the `/r` if you dont want it to be applied to all subfolders (recursive)
 `for /r %%I in (*%Source%) do (`
- Change this line with your ImageMagick command
`"%Convert%" "%InputFile%" "%OutputFile%"`
- Save the code as **ScriptName.bat**
- Run the script in your desired directory.

**Reply by 0196071** [reply]
thanks! if i ever see a decrease in quality with mogrify i will try using this
